### PR TITLE
Add support for "probe" setting

### LINKED ISSUE #36: No support for "probe" setting
vscode-eslint has a `probe` setting:

```js
"eslint.probe": {
    "scope": "resource",
    "type": "array",
    "items": {
        "type": "string"
    },
    "default": [
        "javascript",
        "javascriptreact",
        "typescript",
        "typescriptreact",
        "html",
        "vue",
        "markdown"
    ],
    "description": "An array of language ids for which the extension should probe if support is installed."
},
```

- That setting is processed by the extension and not the server.
- The probe list is checked against the currently open document and based on that the extension decides whether the server will receive notifications about the file or not.
- Based on the result of the previous check, the `validate` option is set to either `on`, `off`, or `probe`
- If `validate` is set to `probe` then the server will:
  - not report errors related to failure to load the eslint library if that's the case
  - if eslint library is not missing then it will apply some default settings based on file extension so that even if there is no proper eslint configuration in the project, the basic validation will work
- Currently, we are hardcoding `validate` to `on` so eslint is unconditionally enabled for all files that match the list of our syntaxes/selectors.
- Eslint extension validates the document against the probe list during handling of:
  - `workspace/configuration`
  - `didOpen` hook
- The server will send a `eslint/probeFailed` message if probing has resulted with setting `validate` to `off` so that client can cache the result

### PR DESCRIPTION
**Pull request by rchl**
Resolves #36

Depends on https://github.com/sublimelsp/LSP/pull/1355

### CODE REVIEW

File: plugin.py [hunk 1]
```diff
@@ -45,6 +59,12 @@ def on_workspace_configuration(self, params, configuration) -> None:
                 if workspace_folder:
                     configuration['workspaceFolder'] = workspace_folder.to_lsp()
                 self.resolve_working_directory(configuration, scope_uri, workspace_folder)
+                buf = session.get_session_buffer_for_uri_async(scope_uri)
+                if buf:
+                    configuration['validate'] = self.compute_validate(buf.language_id, scope_uri, configuration)
+                else:
+                    configuration['validate'] = 'on'
+                del configuration['probe']
```
Comment (rwols) [inline plugin.py:67]: Why is this `del` needed?

Comment (rchl) [reply plugin.py:67]: It's not needed but also the server doesn't need this setting. I've done it to save a few bytes during transfer.
I'm fine with removing this line if you insist. :)
I know deleting properties usually doesn't play nice with compilers and internal optimizations but at the same time this is not a hot path.

Comment (rwols) [reply plugin.py:67]: I don't mind either way, was just confused about it :)

File: plugin.py [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 import re
 import sublime
 import webbrowser
 
-from LSP.plugin.core.url import uri_to_filename
-from lsp_utils import NpmClientHandler
+EslintValidateOn = Literal['on']
+EslintValidateOff = Literal['off']
+EslintValidateProbe = Literal['probe']
+EslintValidate = Union[EslintValidateOn, EslintValidateOff, EslintValidateProbe]
```
Comment (rwols) [inline plugin.py:13]: From the mypy docs about Literal:

> Literals containing two or more values are equivalent to the union of those values. So, Literal[-3, b"foo", MyEnum.A] is equivalent to Union[Literal[-3], Literal[b"foo"], Literal[MyEnum.A]]. This makes writing more complex types involving literals a little more convenient.

I think we can write `Literal['on', 'off', 'probe']` then?

Comment (rchl) [reply plugin.py]: cool, thanks

**Pull request merged**